src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.Helpers.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json
+{
+    public ref partial struct Utf8JsonWriter
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ValidatePropertyNameAndDepth(ref ReadOnlySpan<char> propertyName)
+        {
+            if (propertyName.Length > JsonConstants.MaxCharacterTokenSize || CurrentDepth >= JsonConstants.MaxWriterDepth)
+                ThrowHelper.ThrowJsonWriterOrArgumentException(propertyName, _currentDepth);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ValidatePropertyNameAndDepth(ref ReadOnlySpan<byte> propertyName)
+        {
+            if (propertyName.Length > JsonConstants.MaxTokenSize || CurrentDepth >= JsonConstants.MaxWriterDepth)
+                ThrowHelper.ThrowJsonWriterOrArgumentException(propertyName, _currentDepth);
+        }
+
+        private void ValidateWritingProperty()
+        {
+            if (!_inObject)
+            {
+                Debug.Assert(_tokenType != JsonTokenType.StartObject);
+                ThrowHelper.ThrowJsonWriterException(ExceptionResource.CannotWritePropertyWithinArray, tokenType: _tokenType);
+            }
+        }
+
+        private int WritePropertyNameMinimized(ref ReadOnlySpan<byte> escapedPropertyName)
+        {
+            int idx = 0;
+            if (_currentDepth < 0)
+            {
+                while (_buffer.Length <= idx)
```

Comment:
If I look at the GrowAndEnsure method:
https://github.com/dotnet/corefx/blob/9bec4782491ea4d782d0385956de723930e8b5a6/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs#L780-L793

this one will always Grow, otherwise it will throw, so the while() should not be necessary as the idx is 0, so it will only return if at least one value fits into the buffer.